pkg/apis/messaging/v1/subscribable_channelable_validation.go

```diff
@@ -32,11 +34,20 @@ func isChannelEmpty(f duckv1.KReference) bool {
 func isValidChannel(ctx context.Context, f duckv1.KReference) *apis.FieldError {
 	errs := f.Validate(ctx)
 
-	// Namespace field is disallowed
-	if f.Namespace != "" {
-		fe := apis.ErrDisallowedFields("namespace")
-		fe.Details = "only name, apiVersion and kind are supported fields"
-		errs = errs.Also(fe)
+	if feature.FromContext(ctx).IsEnabled(feature.CrossNamespaceEventLinks) {
+		var resourceInfo cn.ResourceInfo
+		var flag *feature.Store
```

Comment:
@yijie-04 could you set the correct resource info and feature flag variables required for your function to work here?